### PR TITLE
Include docs in sdist and add Makefile for common build, dev, test

### PR DESCRIPTION
**Pull request by benjaoming**
Taking over from #4526

# Change affecting 0.15 release

Docs are included in the sdist and built automatically by running `make sdist`.

PR for #4348 (partly) and #4375

# Change affecting build process

The PyPi and .deb for 0.15 will be released by using a new Makefile. Yay!

#4459 

Makefile does this:

```
➜  ka-lite git:(makefile-dist-docs) make help
clean - remove all build, test, coverage and Python artifacts
clean-build - remove build artifacts
clean-pyc - remove Python file artifacts
clean-test - remove test and coverage artifacts
lint - check style with pep8
test - run tests the default Python
test-bdd - run BDD tests only
test-nobdd - run non-BDD tests only
assets - build all JS/CSS assets
coverage - check code coverage quickly with the default Python
docs - generate Sphinx HTML documentation, including API docs
release - package and upload a release
dist - package locally
install - install the package to the active Python's site-packages

```

### CODE REVIEW

**Comment by benjaoming** [comment]
Still creating the same weird weird `pkg_resources.DistributionNotFound: ka-lite==0.15.0`

**Comment by MCGallaspy** [comment]
Can you rebase against 0.15.x?

File: kalite/distributed/features/steps/content_rating.py [hunk 1]
```diff
@@ -169,17 +167,16 @@ def rate_id(context, id_, val=3):
...
-            rating_el = [el for el in els if int(el.get_attribute("data-val")) == val].pop()
-            rating_el.click()
-            return True
+            return [el for el in els if int(el.get_attribute("data-val")) == val].pop()
         except (NoSuchElementException, StaleElementReferenceException):
             return False
 
-    WebDriverWait(context.browser, 10).until(rate_element)
+    rating_el = WebDriverWait(context.browser, 10).until(get_rating_el)
+    rating_el.click()
```
Comment (MCGallaspy) [inline kalite/distributed/features/steps/content_rating.py]: This is a regression.

Comment (MCGallaspy) [inline kalite/distributed/features/steps/content_rating.py]: We should prefer HEAD here.

File: kalite/distributed/static/js/distributed/rating/views.js [hunk 2]
```diff
@@ -56,20 +56,18 @@ module.exports = BaseView.extend({
...
         var self = this;
-        var callback = function() {
-            self.star_view_difficulty = self.add_subview(StarView, {title: gettext("Difficulty"), el: self.$("#star-container-difficulty"), model: self.model, rating_attr: "difficulty", label_values: this.difficulty_label_values});
-            self.$(".rating-delete").show();
-        };
+
         // If the "quality" is already set, display "difficulty" immediately. Otherwise wait.
-        if (!this.model.get("quality") || parseInt(this.model.get("quality")) === 0) {
-            this.listenToOnce(this.model, "change:quality", callback);
+        if (parseInt(this.model.get("quality")) === 0) {
```
Comment (MCGallaspy) [inline kalite/distributed/static/js/distributed/rating/views.js]: The changes to this file are also a regression.

File: kalite/updates/management/commands/languagepackdownload.py [hunk 3]
```diff
@@ -160,11 +159,17 @@ def move_dubbed_video_map(lang_code):
             logging.error("Error removing dubbed video directory (%s): %s" % (dubbed_video_dir, e))
 
 def move_video_sizes_file(lang_code):
-    """
-    This is no longer needed. See:
-    https://github.com/learningequality/ka-lite/issues/4538#issuecomment-144560505
-    """
-    return
+    lang_pack_location = os.path.join(settings.USER_WRITABLE_LOCALE_DIR, lang_code)
```
Comment (MCGallaspy) [inline kalite/updates/management/commands/languagepackdownload.py]: This also appears to be a regression, but I'm not sure. (I wrote the patches for the above two cases, hence my certainty.)

File: kalitectl.py [hunk 4]
```diff
@@ -569,8 +569,6 @@ def start(debug=False, watch=False, daemonize=True, args=[], skip_job_scheduler=
     except KeyboardInterrupt:
         # Handled in cherrypy by waiting for all threads to join
         pass
-    except SystemExit:
```
Comment (MCGallaspy) [inline kalitectl.py]: Could this also be a regression?

**Comment by benjaoming** [comment]
Hmm, not so nice, sorry about that, I wanted to squash everything into one commit, but perhaps in doing that, I failed to update properly. Will have a look....

**Comment by benjaoming** [comment]
@MCGallaspy thanks for checking this!

File: setup.py [hunk 5]
```diff
@@ -162,6 +171,11 @@ def gen_data_files(*dirs):
     gen_data_files('static-libraries')
 )
 
+data_files += map(
+    lambda x: (os.path.join(kalite.ROOT_DATA_PATH, x[0]), x[1]),
+    gen_data_files('docs/_build/html', optional=True)
```
Comment (MCGallaspy) [inline setup.py]: Use `os.path.join('docs', '_build', 'html')`

**Pull request merged**